### DISCUSSION TITLE
Some questions about the future of AIR.

### DISCUSSION THREAD

**Discussion by Jason0247** [airsdk/Adobe-Runtime-Support#1243] [Q&A]
Flashplayer has stopped updating. I'd like to know whether this will affect the development of desktop applications and mobile applications in the future, and whether there will be an association between flashplayer and air.
Many previous projects were based on swf format and embedded into c# or another platform, because the effect of SWF is really cool and convenient. With the update of windows 10, the system keeps prompting updates or reporting error messages, resulting in that flash player can no longer run normally, or it can only run if the operation is very troublesome. Therefore, there has always been a question whether the air will suddenly stop

**Comment by ajwfrost** [comment]
Flash Player was designed as a browser plug-in. Embedding it into other C#/etc applications via ActiveX is something that a lot of folk did and it was convenient, yes, but was never an officially supported use case by Adobe.

Ultimately, the decision to stop supporting the Flash Player was driven by the browser environments where they had their own capabilities and were trying to get people to use standards-based approaches for rich web-based content, plus there was a potential for security concerns because of this natively written plug-in running in its own fashion on a desktop machine (although they all ended up with a sandbox around it to a certain extent..).

So then it's just a by-product of the fact that the browsers didn't want to support plug-ins that means Adobe stopped supporting the Flash Player - and due to the potential exposure if there had been security flaws exposed post-support, Adobe and Microsoft took steps to ensure that the Flash Player was disabled and/or removed from machines.

So it's a very different situation with AIR! It's a platform designed to run as standalone applications on a desktop so the security model is very different and the use cases are completely different. Whilst it's not as popular as it used to be, and has probably lost out due to the removal of Flash Player, it's still widely used for some surprisingly popular/prevalent games and applications. I've just been putting together a summary for our management on how things have been going since we took this over and started publishing the v33 SDKs, and we are still seeing continued growth in take-up of our commercial license tiers which is encouraging. So it should have life for quite a while yet!

In terms of it 'suddenly stopping' - there is no need for any sort of disablement like Adobe put in for the Flash Player, because of the different environment/ecosystem that AIR apps run in, so that won't happen. The worst case would be if we stopped supporting it too, at which point (a) anything you had you could still use (the AIR SDK license is perpetual); (b) things would perhaps slowly start to be tricky to work around i.e. new requirements from Android/iOS, but otherwise it could still be used. Plus, if Harman did decide to stop supporting it, the whole code-based could be picked up and handed to someone else to take over (it's still owned by Adobe at the end of the day).

Regarding those apps that you had which were embedding the Flash Player into C# applications .. I can't see any reason why AIR couldn't expose an ActiveX control interface for some content so that the same effect could be achieved. But also, there are a number of application vendors who have licensed the Flash Player from us so that they can continue to use and ship this with their applications, we are still supporting the Flash Player as well as AIR but only for enterprise software vendors rather than as a public offering.

Hope that helps...!

**Reply by 2jfw** [reply]
> Plus, if Harman did decide to stop supporting it, the whole code-based could be picked up and handed to someone else to take over (it's still owned by Adobe at the end of the day).

Just wondering, if Adobe will ever _fully_ give away/sell AIR - and also if not, why? Because of Animate? Do they still earn _so_ much with the licensing of AIR? I think the future of AIR would be even brighter if they did, also giving Harman the chance to go even deeper with improvements and new features..

**Reply by ajwfrost** [reply]
Do you mean, to open source it? That won't happen at least while Flash Player is still a thing, they share enough codebase that the security implications require a high level of confidentiality of the source code... which is one reason why the software is just licensed to us. Also, there are lots of other bits of IPR in the codebase that are shared with other Adobe products (e.g. font rendering, video solutions etc) so it's just a little messy!

**Reply by 2jfw** [reply]
> Do you mean, to open source it? That won't happen at least while Flash Player is still a thing, they share enough codebase that the security implications require a high level of confidentiality of the source code... which is one reason why the software is just licensed to us. Also, there are lots of other bits of IPR in the codebase that are shared with other Adobe products (e.g. font rendering, video solutions etc) so it's just a little messy!

No, not open-sourcing it but completely handing AIR over to you - thanks for the details here, I now can see the reason for adobe not doing this when their codebase security is critical and based on other products code etc.

**Comment by Jason0247** [comment]
@ajwfrost 

Thank you for dispelling my doubts and believing that the future of air will become more wonderful.

It is expected that air will have more and more functions and can obtain a higher share in the mobile market. So far, I always feel that air still lacks a powerful editing environment, like Xcode. Perhaps in the future, there will be a powerful editor that is only used by air, and air will go further. After all, there are still many as developers. Even after many years, if there is such an editing environment, I believe many people will consider returning, because the development of air is really too convenient

**Reply by dgurney75** [reply]
Adobe Animate is the most powerful all-in-one IDE that is used to develop Adobe AIR applications. Leaps and bounds better than Xcode in my opinion.

**Reply by Jason0247** [reply]
Animate is also the IDE I have been using for developing desktop applications. It is indeed a good ide. More accurately, it is a good platform for creating animation.

However, on the mobile end, animate is really not a development platform suitable for air. Compared with flex, flex is better, but it is not updated, and it is really unfriendly and too bloated in skin management.

**Reply by 2jfw** [reply]
Just pointing out to **Moonshine** IDE, made for as3/AIR/Flex/Royale development or **IntelliJ**, one of the best professional IDEs, perfectly supporting as3/AIR/Flex development.
